sdk/Sdk.Generators/FunctionMetadataProviderGenerator.Parser.cs

```diff
@@ -517,6 +517,33 @@ private bool TryGetAttributeProperties(AttributeData attributeData, Location att
                     }
                 }
 
+                // some properties have default values, so if these properties were not already defined in constructor or named arguments, we will auto-add them here
+                foreach (var member in attributeData.AttributeClass!.GetMembers().Where(a => a is IPropertySymbol))
+                {
+                    var defaultValAttrList = member.GetAttributes().Where(attr => SymbolEqualityComparer.Default.Equals(attr.AttributeClass, Compilation.GetTypeByMetadataName(Constants.DefaultValueType)));
+
+                    if (defaultValAttrList.SingleOrDefault() is { } defaultValAttr) // list will only be of size one b/c there cannot be duplicates of an attribute on one piece of syntax
+                    {
+                        var argName = member.Name;
+                        var argDefaultVal = defaultValAttr.ConstructorArguments.SingleOrDefault().Value!.ToString(); // only one constructor arg in DefaultValue attribute (the default value)
```

Comment:
The value can be a string or a bool here. Would pattern matching fit better for branching?

``` CSharp
object arg = defaultValAttr.ConstructorArguments.SingleOrDefault().Value!;
switch arg
{
    case bool b:
        // used bool constructor
    case string s:
        // use string constructor
    defaullt:
        throw new NotSupportedException(); // used unknown ctor.
}
```

Reply:
The dictionary created here is <string, object> so I don't think we need to differentiate much - I revised this to just take the value as a generic object and set the dictionary value to that. 

The only special handling we need in this spot is to change the IsBatched value to "Cardinality" for the function metadata info (which I've also revised slightly in this iteration to use arg as a bool).